### DISCUSSION TITLE
What HW setup u ppl recommend

### DISCUSSION THREAD

**Discussion by jaroslavhuss** [lllyasviel/Fooocus#1075] [Q&A]
Hey, 

I am on a MacBook Pro M1 (16gb ram) and generating an image takes forever. About several hours. What machine setup u guys recommend? 

Thank you!

**Comment by xhoxye** [comment]
You can try LCM and turbo

**Comment by realmalikov** [comment]
I myself have a slightly older 1060 with 6GB VRAM and have found a model based on SDXL Turbo that reduces the generation time by more than 60% (under a minute!). These are my settings for fast and good results:

Model: https://civitai.com/models/129666/realities-edge-xl-lcmsdxlturbo
Steps (overwrite in advanced settings): 5 to 6
CFG (overwrite in advanced settings): 1 to 2
Sampler (overwrite in advanced settings): dpmpp_sde_gpu (IMPORTANT)

**Comment by mel870** [comment]
I have a Macbook Pro M1 Max 32GB, 32GPU. Fooocus really didn't work well for fast generations, so I had to give up trying. For Mac, I use an app called Draw Things. It uses Core ML, neural, etc to help with speed, but I still prefer Fooocus. So I use Draw Things to hammer out ideas (for free). Then, I use Fooocus, Comfy, or Auto1111 on a rented 3090 or 4090 with RunPod or Tensordock to finalize images. On either service, I can rent a GPU for less than $0.50 an hour. I also backup my entire setup to Backblaze and restore to save setup time since sometimes the same GPU's may not be available and you otherwise have to setup everything again with no backup. 

LCM models are good as well as stated above, but if you like trying other models that don't yet support it, renting a GPU is best.

**Reply by realJogicodes** [reply]
$0.5/hour is a great price, better than what I could find. Where's that? I've got a laptop with intel graphics and that solution would work for me

**Reply by mel870** [reply]
I use Tensor or Runpod depending on what's available: 3090 is about $0.30 on Tensor. I also like that you can customize all specs on Tensor but lately they have had no GPU's except A6000 or higher which is overkill for running just Fooocus but even an A6000 is about $0.55. They also have 16GB GPU as well. RunPod also has cheap GPU's if you use community cloud. 

![Screenshot 2023-11-30 at 5 20 55 PM](https://github.com/lllyasviel/Fooocus/assets/7969647/9507f75e-8c8d-45e7-8a9b-930cefdf71d6)


[Tensor Dock ](https:///marketplace.tensordock.com)
[Run Pod](https://www.runpod.io/)

**Reply by realJogicodes** [reply]
Thank you. That's dope. I have run out of the free google colab compute allowance and this pricing is amazingly cheap. I imagine that can run other models too. 

Is that experience similar to the google colab where the user facing front-end is a graphical browser "instance"

**Reply by mel870** [reply]
With RunPod and Tensor you have the option to use Jupyter lab. So pretty much the same UI as Colab.  Runpod is a lot more user friendly if you're not comfortable using Terminal since they have docker templates for Stable Diffusion, and you can probably find one for Fooocus as well on https://hub.docker.com/. This week, all 3090's and 4090's have been in use on Tensor. They are working on adding more providers. In the meantime I'm using RunPod.

**Comment by Iann-urus** [comment]
On Fri, 1 Dec 2023 at 01:54, Jogi ***@***.***> wrote:

> $0.5/hour is a great price, better than what I could find. Where's that?
> I've got a laptop with intel graphics and that solution would work for me
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lllyasviel/Fooocus/discussions/1075#discussioncomment-7723107>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AP7PURB3JXCC5KVF6X6GPI3YHEFDRAVCNFSM6AAAAABAAYRENSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOMRTGEYDO>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
I’d like to know as well
